exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/efgs/FederationGatewaySyncProcessor.java

```diff
@@ -17,36 +16,43 @@
 public class FederationGatewaySyncProcessor {
 
     private static final Logger LOG = LoggerFactory.getLogger(FederationGatewaySyncProcessor.class);
+    private LocalDate lastInboundSyncFromEfgs;
```

Comment:
To maintain the class as thread-safe, I'd throw this into a private final AtomicReference<LocalDate>. Of course in practice, it likely won't ever matter here, but in theory, the next pass might not be done by the same thread.

Reply:
Yep, that is true, but even if it's run more than once as parallel, the value is only updated more than once. However, there might be some visibility issues for updates, so good point.

Reply:
However, I checked that we have only @EnableScheduling defined in our configuration, so it seems that our scheduler is single-threaded. So, not much problem of concurrency then. :) Anyway, maybe we should define some thread pool because we have increasing count of jobs. Lets talk about this tomorrow.

_By default, will be searching for an associated scheduler definition: either a unique TaskScheduler bean in the context, or a TaskScheduler bean named "taskScheduler" otherwise; the same lookup will also be performed for a ScheduledExecutorService bean. If neither of the two is resolvable, a local single-threaded default scheduler will be created and used within the registrar._

Reply:
Although, our services seems not to be very timing related, so single thread might actually be enough. Just thinking if there is a risk that it get stuck. However, running more threads is a bad solution anyway. Anyway, we could add something like this in future in application.yml if there is need for that:
`spring.task.scheduling.pool.size=5`

Reply:
Yeah, a single scheduler should be fine for us. Only reason for adding more would be if for example the efgs part starts taking so long that we don't want other tasks (like cleaning keys from db) to wait for it. I don't think there's much risk of that. None the less, in services like this where everything is 99% stateless anyhow, I like to assume that I don't know the threading/call orders -> avoid state first and thread-safe the small bits that you might find needed.